### PR TITLE
Add scripts and Makefile targets to automatically update AMI IDs

### PR DESCRIPTION
**Pull request by webvictim**
As per PRs like https://github.com/gravitational/teleport/pull/4169, updating the AMI IDs used in the Cloudformation YAML files is currently a manual process. This is fairly trivial to automate with the use of a script and the `aws` CLI tool.

Current workflow:
- Update `TELEPORT_VERSION` in `assets/aws/Makefile`
- Run `make -C assets/aws update-ami-ids-cloudformation assets/aws update-ami-ids-terraform`
- Commit the result to a branch and raise a PR

Referenced in #4031

### CODE REVIEW

File: assets/aws/update-ami-ids.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    done
+elif [[ "${MODE}" == "terraform" ]]; then
+    TERRAFORM_PATH=../../examples/aws/terraform/README.md
+    if [[ "${TYPE}" == "oss" ]]; then
+        TYPE_STRING="OSS"
+    elif [[ "${TYPE}" == "ent" ]]; then
+        TYPE_STRING="Enterprise"
+    elif [[ "${TYPE}" == "ent-fips" ]]; then
+        TYPE_STRING="Enterprise FIPS"
+    fi
```
Comment (benarent) [inline assets/aws/update-ami-ids.sh:92]: Nice!

**Review by benarent — APPROVED** [review]

File: assets/aws/Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ MARKETPLACE_AMI_NAME ?=
 AWS_REGION ?= us-west-2
 
 # Teleport version
-TELEPORT_VERSION ?= 4.3.2
+TELEPORT_VERSION ?= 4.3.5
```
Comment (awly) [inline assets/aws/Makefile:17]: Add a comment saying that this should be the last _released_ version available to the public.
Prevents someone from matching it to an unreleased value from the root `Makefile`

File: assets/aws/update-ami-ids.sh [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+set -e
```
Comment (awly) [inline assets/aws/update-ami-ids.sh:2]: nit: `set -euo pipefail`

File: assets/aws/update-ami-ids.sh [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        TYPE_STRING="Enterprise FIPS"
+    fi
+    # replace AMI ID in place
+    for REGION in ${REGIONS//,/ }; do
+        OLD_AMI_ID=$(grep -E "# $REGION v(.*) ${TYPE_STRING}" $TERRAFORM_PATH | sed -n -E "s/# $REGION v(.*) ${TYPE_STRING}: (ami.*)/\2/p" | tr -d " ")
+        NEW_AMI_ID=${IMAGE_IDS[$REGION]}
+        sed -i -E "s/^# $REGION v(.*) ${TYPE_STRING}: ami(.*)$/# $REGION v${VERSION} ${TYPE_STRING}: $NEW_AMI_ID/g" $TERRAFORM_PATH
+        echo "[${TYPE}: ${REGION}] ${OLD_AMI_ID} -> ${NEW_AMI_ID}"
+    done
+fi
```
Comment (awly) [inline assets/aws/update-ami-ids.sh:102]: nit: missing new line at the end of file

File: examples/aws/terraform/README.md [hunk 5]
```diff
@@ -66,3 +66,65 @@ export TF_VAR_grafana_pass="CHANGE_THIS_VALUE"
 # plan
 make plan
 ```
+
+## Public Teleport AMI IDs
```
Comment (awly) [inline examples/aws/terraform/README.md:70]: Will these also get automatically updated?

**Review by awly — APPROVED** [review]

**Pull request merged**